src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4006,6 +4006,36 @@ GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
     return andnNode;
 }
 
+void Lowering::LowerBswapOp(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_BSWAP));
+
+    if (!comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE))
+    {
+        return;
+    }
+
+    GenTree* operand = node->gtGetOp1();
+    if (operand->OperIs(GT_IND))
+    {
+        operand->SetContained();
+
+        return;
+    }
+
+    LIR::Use use;
+    if (!BlockRange().TryGetUse(node, &use))
```

Comment:
Ehhhhh, I am personally not a big fan of using `TryGetUse` for peepholes (for TP reasons of course).

Would this look too unsightly if done when containment-checking the user (`STOREIND`)?

Reply:
My reasoning for this was that the BSWAP node is really rare and that the general impact on TP is low and the benefit is more simplified check.

Reply:
> BSWAP node is really rare and that the general impact on TP is low

Yes, that I agree with. My concern is with people following patterns (/copying code), and this is not a "good" pattern.

I think if we leave it as-is, we should also have a comment stating that we are using `TryGetUse` explicitly because we expect `BSWAP`s to be rare.

Reply:
I agree that we should move this handling to `GT_STOREIND`. In general we should not be modifying future nodes that will be lowered, especially not with non-trivial changes like reordering.